### ISSUE TITLE
Color control dynamic tag

### ISSUE THREAD

**Issue by drewkarriker** [elementor/elementor#6485]
>>>I am trying to use ACF pro to create a custom field type for colors, ACF Pro comes with a color picker. I would like it to make a custom field that asks for primary, secondary, font, and accent, as well as other custom ones. I have done this but my question is how do I make those color selections within a custom field auto propagate into Elementor's color picker? Is there a dynamic (pull from a post-meta/acf field) setting I am missing on the color picker? If this is not something you offer can I make a product suggestion to make this happen and/or can you suggest a plugin/integration or work flow that will allow this?

**Comment by bainternet** [comment]
Currently, this feature is not supported, but we will look into it in the future if we see a large demand for it.

Thanks for the feedback.

**Comment by teolaz** [comment]
@bainternet why closing feature requests if they're not implemented yet?...
+1 here

**Comment by papilia** [comment]
+1

**Comment by philmatt** [comment]
+1

The _color_picker_ field is listed as a supported field by Elementor, but doesn't seem to be integrated in any useful fashion. Is this on the roadmap?

@shilo-ey can you please reopen?

**Comment by Tvhawk** [comment]
+1

**Comment by shilo-ey** [comment]
[**@philmatt**](https://github.com/philmatt) commented on [Feb 11, 2019, 9:01 AM GMT+2](https://github.com/elementor/elementor/issues/6485#issuecomment-462233594 "2019-02-11T07:01:29Z - Replied by Github Reply Comments"):
> +1
> 
> The _color_picker_ field is listed as a [supported field by Elementor](https://docs.elementor.com/article/381-elementor-integration-with-acf), but doesn't seem to be integrated in any useful fashion. Is this on the roadmap?
> 
> @shilo-ey can you please reopen?

Please note that this field is under "Text", which means that it retrieves a string with the color value.

**Comment by drewb1** [comment]
@shilo-ey I think you misunderstand.  We want the color picker on the site to have a dynamic function that pulls from a custom field.  
Please reopen!

**Comment by Tvhawk** [comment]
@shilo-ey - We want to be able to use the ACF Color Picker fields to dynamically fill things like this: http://media.hawks.link/992462

I.e. we set up a 5 page template site for use with budget clients. We can set up an acf options page - set the brand colors, accent colors, etc and use these options to dynamically swap out the colors across this 5 page site without having to go to each and every element to adjust colors.

The alternative way of doing this would be using CSS Variables, but that's also another unsupported feature.

**Comment by fracii** [comment]
+1

**Comment by Tvhawk** [comment]
I just added this as a poll option on this poll: https://www.facebook.com/groups/Elementors/permalink/1918292704963173/

Everyone here should go vote for it.

**Comment by tristanleboss** [comment]
+1

**Comment by drewb1** [comment]
@bainternet please re-open

**Comment by fredarend** [comment]
@bainternet please re-open

**Comment by black-eye** [comment]
+1

**Comment by nuphunk** [comment]
+1

**Comment by Fanny8p** [comment]
+1 Please re-open !!!

**Comment by tristanleboss** [comment]
They don't read comments. The best way to have them reconsider this issue is to open a new one linking to this one.

**Comment by drewb1** [comment]
> They don't read comments. The best way to have them reconsider this issue is to open a new one linking to this one.

https://github.com/elementor/elementor/issues/8130

**Comment by black-eye** [comment]
> > They don't read comments. The best way to have them reconsider this issue is to open a new one linking to this one.
> 
> #8130

They will close your new issue either. You have to fill in the issue properly, with al the checkboxes checked and all the points answered. They don't understand human speech... ;)

**Comment by bainternet** [comment]
[**@tristanleboss**](https://github.com/tristanleboss) commented on [May 17, 2019, 3:53 AM GMT+3](https://github.com/elementor/elementor/issues/6485#issuecomment-493280021 "2019-05-17T00:53:09Z - Replied by Github Reply Comments"):
> They don't read comments. The best way to have them reconsider this issue is to open a new one linking to this one.

Wrong buddy 🤪

**Comment by drewb1** [comment]
THEY REOPENED IT!!!!!

**Comment by tristanleboss** [comment]
@bainternet Maybe you can reopen this one too: https://github.com/elementor/elementor/issues/7552
Converting the form redirect URL field to a dynamic one is probably a 5 minute job and so logical! By this, I mean why should this field be a text field if its main purpose is to contain an URL? Why can't this field benefits from the autocomplete and dynamic feature that all other URL fields accros Elementor benefits?

**Comment by dolfje** [comment]
It is rather easy to implement because of the very good coding style of this plugin!!
I enabled it for the call to action module (hover overlay). But you can add it in any place that have a color control. See code below.

```
Elementor module
================
diff --git a/includes/controls/color.php b/includes/controls/color.php
index 29a86ac..4749b3b 100644
--- a/includes/controls/color.php
+++ b/includes/controls/color.php
@@ -1,6 +1,8 @@
 <?php
 namespace Elementor;
 
+use Elementor\Modules\DynamicTags\Module as TagsModule;
+
 if ( ! defined( 'ABSPATH' ) ) {
        exit; // Exit if accessed directly.
 }
@@ -86,7 +88,7 @@ class Control_Color extends Base_Data_Control {
                                        <span class="elementor-control-field-description">{{{ data.description }}}</span>
                                <# } #>
                        </label>
-                       <div class="elementor-control-input-wrapper">
+                       <div class="elementor-control-input-wrapper elementor-control-tag-area">
                                <input data-setting="{{ name }}" type="text" placeholder="<?php echo esc_attr( 'Hex/rgba', 'elementor' ); ?>" {{ defaultValue }}{{ dataAlpha }} />
                        </div>
                </div>
@@ -108,6 +110,9 @@ class Control_Color extends Base_Data_Control {
                return [
                        'alpha' => true,
                        'scheme' => '',
+                        'dynamic' => [
+                                'categories' => [ TagsModule::TEXT_CATEGORY ],
+                        ],
                ];
        }
 }

Elementor Pro Module
====================
diff --git a/modules/call-to-action/widgets/call-to-action.php b/modules/call-to-action/widgets/call-to-action.php
index b53a89d..84fd336 100644
--- a/modules/call-to-action/widgets/call-to-action.php
+++ b/modules/call-to-action/widgets/call-to-action.php
@@ -1396,6 +1396,9 @@ class Call_To_Action extends Base_Widget {
                                'selectors' => [
                                        '{{WRAPPER}} .elementor-cta:not(:hover) .elementor-cta__bg-overlay' => 'background-color: {{VALUE}}',
                                ],
+                               'dynamic' => [
+                                       'active' => true,
+                               ],
                        ]
                );
 
@@ -1450,6 +1453,9 @@ class Call_To_Action extends Base_Widget {
                                'selectors' => [
                                        '{{WRAPPER}} .elementor-cta:hover .elementor-cta__bg-overlay' => 'background-color: {{VALUE}}',
                                ],
+                                'dynamic' => [
+                                        'active' => true,
+                                ],
                        ]
                );
 
```

**Comment by julien-jacob** [comment]
+1 ;)

**Comment by logobby** [comment]
I'm also need this feature but meanwhile I want to share with you an AMAZING plugin that can resolves you this issue (depending on what you want to achiev) and also let you do a ton of things (create your own Elementor Widgets easily and free...).

Link (affiliate) to the plugin: http://bit.ly/unlimited-elements-for-elementor enjoy it and good work :)

**Comment by tristanleboss** [comment]
Direct link: https://unlimited-elements.com/

**Comment by philmatt** [comment]
+1
The lack of this feature is holding back my website development. Please enable this!

**Comment by luiszg** [comment]
Hello I need it also.

**Comment by ProtoThemes** [comment]
+1

**Comment by Stevens1305** [comment]
+1

This is a no brainer. Would make sitebuilding so much easier and bring a broader range of customisation options

**Comment by tollube** [comment]
> It is rather easy to implement because of the very good coding style of this plugin!!
> I enabled it for the call to action module (hover overlay). But you can add it in any place that have a color control. See code below.
> 
> edit: It also works with rgba values if you install https://github.com/reyhoun/acf-rgba-color
> 
> ```
> Elementor module
> ================
> diff --git a/includes/controls/color.php b/includes/controls/color.php
> index 29a86ac..4749b3b 100644
> --- a/includes/controls/color.php
> +++ b/includes/controls/color.php
> @@ -1,6 +1,8 @@
>  <?php
>  namespace Elementor;
>  
> +use Elementor\Modules\DynamicTags\Module as TagsModule;
> +
>  if ( ! defined( 'ABSPATH' ) ) {
>         exit; // Exit if accessed directly.
>  }
> @@ -86,7 +88,7 @@ class Control_Color extends Base_Data_Control {
>                                         <span class="elementor-control-field-description">{{{ data.description }}}</span>
>                                 <# } #>
>                         </label>
> -                       <div class="elementor-control-input-wrapper">
> +                       <div class="elementor-control-input-wrapper elementor-control-tag-area">
>                                 <input data-setting="{{ name }}" type="text" placeholder="<?php echo esc_attr( 'Hex/rgba', 'elementor' ); ?>" {{ defaultValue }}{{ dataAlpha }} />
>                         </div>
>                 </div>
> @@ -108,6 +110,9 @@ class Control_Color extends Base_Data_Control {
>                 return [
>                         'alpha' => true,
>                         'scheme' => '',
> +                        'dynamic' => [
> +                                'categories' => [ TagsModule::TEXT_CATEGORY ],
> +                        ],
>                 ];
>         }
>  }
> 
> Elementor Pro Module
> ====================
> diff --git a/modules/call-to-action/widgets/call-to-action.php b/modules/call-to-action/widgets/call-to-action.php
> index b53a89d..84fd336 100644
> --- a/modules/call-to-action/widgets/call-to-action.php
> +++ b/modules/call-to-action/widgets/call-to-action.php
> @@ -1396,6 +1396,9 @@ class Call_To_Action extends Base_Widget {
>                                 'selectors' => [
>                                         '{{WRAPPER}} .elementor-cta:not(:hover) .elementor-cta__bg-overlay' => 'background-color: {{VALUE}}',
>                                 ],
> +                               'dynamic' => [
> +                                       'active' => true,
> +                               ],
>                         ]
>                 );
>  
> @@ -1450,6 +1453,9 @@ class Call_To_Action extends Base_Widget {
>                                 'selectors' => [
>                                         '{{WRAPPER}} .elementor-cta:hover .elementor-cta__bg-overlay' => 'background-color: {{VALUE}}',
>                                 ],
> +                                'dynamic' => [
> +                                        'active' => true,
> +                                ],
>                         ]
>                 );
>  
> ```

Hi dolfje!
Where to add that? And how to make it possible for the section background color? I really need that. Thanks in advance!

**Comment by shisan79** [comment]
+1

**Comment by parfilov** [comment]
+1

**Comment by dudaster** [comment]
Meanwhile you can use Attributes to serve your purpouse.

**Comment by Tvhawk** [comment]
> Meanwhile you can use Attributes to serve your purpouse.

huh?

**Comment by chloeford0** [comment]
+1 for a dynamic color picker!

**Comment by gabrieluntura** [comment]
+1 it's a much needed feature

**Comment by LinusOkidok** [comment]
+1 would help out a lot!

**Comment by jflagarde** [comment]
This would be an awesome feature to have. It would allow so much more customizations and speed up our workflow tremendously.

**Comment by momo-fr** [comment]
+100000000000000 I want that! PLEASE!

**Comment by dsgepers** [comment]
+1

**Comment by maxjutras** [comment]
+1

**Comment by PixelPerfect-dot** [comment]
+1 Of course you'll need this

**Comment by iceflow19** [comment]
This would be great to have but I have a somewhat simple workaround.

1. Insert a shortcode widget somewhere in the template.
2. Install a plugin that allows you to wrap php code in a shortcode (Ie. [Insert PHP Code Snippet](https://wordpress.org/plugins/insert-php-code-snippet/))
3. Create a shortcode wrapping `<?php echo the_field('<--your color picker acf field here-->') ?>`
4. Insert the shortcode along with css into the shortcode widget. Ex:
```
<style type="text/css">
#banner {
   background-color: [xyz-ips snippet="ColorField"];
}
</style>
```

**Comment by leem2209** [comment]
> This would be great to have but I have a somewhat simple workaround.
> 
> 1. Insert a shortcode widget somewhere in the template.
> 2. Install a plugin that allows you to wrap php code in a shortcode (Ie. [Insert PHP Code Snippet](https://wordpress.org/plugins/insert-php-code-snippet/))
> 3. Create a shortcode wrapping `<?php echo the_field('<--your color picker acf field here-->') ?>`
> 4. Insert the shortcode along with css into the shortcode widget. Ex:
> 
> ```
> <style type="text/css">
> #banner {
>    background-color: [xyz-ips snippet="ColorField"];
> }
> </style>
> ```

Hey iceflow19, could you elaborate more how to implement this code? I'm struggling to get this to work.

**Comment by koystrubvs** [comment]
+1

**Comment by yansircc** [comment]
+1
It would be really helpful if I can request dynamic color code in my URL parameter and make it as a dynamic color for any background color in my page.

**Comment by kristiangl** [comment]
I'm working on a site at the moment where I needed this. Here is a fairly messy workaround similar to iceflow19's, although I plan on going back later and implementing when the feature has been added (added to the next 2.9 Pro Milestone!). I'm using Pods, so there are already working shortcodes for pulling through the colour code.

**1. Fields added to post-type using Pods**
<img width="1169" alt="Screen Shot 2020-02-13 at 1 18 20 pm" src="https://user-images.githubusercontent.com/4176599/74403954-0b35e900-4e64-11ea-9fc4-51f862fb3539.png">

**3. Inspect the page and figure out the CSS required to change the colours you want**

**3. Add Shortcode Element to page, with required CSS**
In the text field, add the CSS you figured out in step 2, surrounded by `<style>` tags, and replace the colours in the CSS with the pods shortcode corresponding to your field:
<img width="1607" alt="Screen Shot 2020-02-13 at 1 16 41 pm" src="https://user-images.githubusercontent.com/4176599/74404091-78497e80-4e64-11ea-9de3-8ef911190a33.png">

**Comment by Morgunov-Vitaly** [comment]
Yes, **kristiangl**. But, unfortunately, it doesn't help if you want to display list of taxonomies (for example - colored tags)

**Comment by Worldwebscapes** [comment]
We are getting a lot of questions popping up in the Facebook Group about this new dynamic button.

Propose we add a new tooltip to help with these new "Easter Eggs" as we tweak.
Something to this effect.

![image](https://user-images.githubusercontent.com/7572038/75562056-fa65a580-5a15-11ea-9f83-927db2183975.png)

**Comment by spac-em** [comment]
+1 for this. i am using  third party plugin for this now . Epic News Elements

**Comment by shilo-ey** [comment]
This issue has been resolved in Elementor Pro v2.9.1

Feel free to update

Thanks!
